frontend/src/views/admin/matching/types.ts

```diff
@@ -0,0 +1,36 @@
+import {
+    Position,
+    Applicant,
+    Assignment,
+    Application,
+} from "../../../api/defs/types";
+
+export type Match = {
+    applicantId: number;
+    utorid: string;
+    positionId: number;
+    positionCode: string;
+    status: "applied" | "starred" | "staged-assigned" | "assigned" | "hidden";
+    hoursAssigned: number;
+};
+
+export type AppointmentGuaranteeStatus = {
+    utorid: string;
+    totalHoursOwed: number;
+    previousHoursFulfilled: number;
+};
+
+export type PositionSummary = {
+    position: Position;
+    hoursAssigned: number;
+    filledStatus: "empty" | "under" | "matched" | "over";
```

Comment:
You use this elsewhere. You can refer to this specific subtype by doing,
```typescript
const status: PositionSummary["filledStatus"] = ...
```
for example.